### PR TITLE
fix bug #294 (ibm out of memory problem with iibm=3)

### PR DESCRIPTION
**Pull request by slaizet**

### CODE REVIEW

File: src/ibm.f90 [hunk 1]
```diff
@@ -521,7 +522,8 @@ subroutine cubsplx(u,lind)
               endif
               ! Special Case
               if (xi(i,j,k).eq.xf(i,j,k)) then
-                  u(ipol,j,k)=bcimp                                   
+                 call decomp_2d_abort(1, "!! situation not supported by the IBM !!")
+                 !u(ipol,j,k)=bcimp
```
Comment (pbartholomew08) [inline src/ibm.f90:526]: Delete commented code?

File: src/ibm.f90 [hunk 2]
```diff
@@ -688,7 +691,8 @@ subroutine cubsply(u,lind)
               endif
               ! Special Case
               if (yi(j,i,k).eq.yf(j,i,k)) then
-                  u(i,jpol,k)=bcimp                                   
+                 call decomp_2d_abort(1, "!! situation not supported by the IBM !!")
+                 !u(i,jpol,k)=bcimp
```
Comment (pbartholomew08) [inline src/ibm.f90:695]: Delete commented code?

File: src/ibm.f90 [hunk 3]
```diff
@@ -837,28 +842,29 @@ subroutine cubsplz(u,lind)
...
-              !     ! Special Case
-              !     if (zi(k,i,j).eq.zf(k,i,j)) then
-              !         u(i,j,kpol)=bcimp                                   
-              !     else              
+              ! Special Case
+              if (zi(k,i,j).eq.zf(k,i,j)) then
+                 !u(i,j,kpol)=bcimp
+                 call decomp_2d_abort(1, "!! situation not supported by the IBM !!")
+              else              
               ! Cubic Spline Reconstruction
```
Comment (pbartholomew08) [inline src/ibm.f90:850]: Fix comment indentation, otherwise this is confusing to read.

File: src/ibm.f90 [hunk 4]
```diff
@@ -837,28 +842,29 @@ subroutine cubsplz(u,lind)
                     endif
                  enddo
               endif
-              !     ! Special Case
-              !     if (zi(k,i,j).eq.zf(k,i,j)) then
-              !         u(i,j,kpol)=bcimp                                   
-              !     else              
+              ! Special Case
+              if (zi(k,i,j).eq.zf(k,i,j)) then
+                 !u(i,j,kpol)=bcimp
```
Comment (pbartholomew08) [inline src/ibm.f90:847]: Delete commented code?

File: src/ibm.f90 [hunk 5]
```diff
@@ -837,28 +842,29 @@ subroutine cubsplz(u,lind)
...
+                 call decomp_2d_abort(1, "!! situation not supported by the IBM !!")
+              else              
               ! Cubic Spline Reconstruction
-              na=ia
-              do kpol=kpoli,kpolf 
+                 na=ia
+                 do kpol=kpoli,kpolf 
                  ! Special Case
```
Comment (pbartholomew08) [inline src/ibm.f90:853]: As above

**Review by mathrack — APPROVED** [review]

File: src/ibm.f90 (hunk 5 above)
Comment (pbartholomew08) [reply src/ibm.f90:853]: Can we describe the special case to avoid confusion in future?

**Review by pbartholomew08 — CHANGES_REQUESTED** [review]
Minor changes only, I think adding a comment to explain the special case could be particularly beneficial.

File: src/ibm.f90 (hunk 5 above)
Comment (slaizet) [reply src/ibm.f90:853]: happy to, but I do not know what the special case is :-) I was not able to get a special case!

**Pull request merged**